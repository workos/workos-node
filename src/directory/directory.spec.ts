import WorkOS from '../workos';

describe('SSO', () => {
  describe('getAuthorizationURL', () => {
    it('generates an authorize url with the directory sync flag set', () => {
      const workos = new WorkOS('sk_test_Sz3IQjepeSWaI4cMS4ms4sMuU', {
        apiHostname: 'api.workos.dev',
      });

      const url = workos.directory.getAuthorizationURL({
        domain: 'lyft.com',
        projectID: 'proj_123',
        redirectURI: 'example.com/sso/workos/callback',
      });

      expect(url).toMatchSnapshot();
    });
  });
});
