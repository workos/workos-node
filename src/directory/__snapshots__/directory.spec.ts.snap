// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SSO getAuthorizationURL generates an authorize url with the directory sync flag set 1`] = `"https://api.workos.dev/sso/authorize?client_id=proj_123&domain=lyft.com&is_directory_sync=true&redirect_uri=example.com%2Fsso%2Fworkos%2Fcallback&response_type=code"`;
